updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,76 @@
+* removing the custom labels from the remaining nodes in small batches and testing the applications on those nodes 
+
+//The following wording comes from https://github.com/openshift/openshift-docs/pull/34704, not yet finalized
+
+[NOTE]
+====
+Pausing a machine config pool prevents the Machine Config Operator from applying any configuration changes on the associated nodes. Pausing an MCP also prevents any automatically-rotated certificates from being pushed to the associated nodes, including the automatic CA rotation of the `kube-apiserver-to-kubelet-signer` CA certificate. If the MCP is paused when the `kube-apiserver-to-kubelet-signer` CA certificate expires and the MCO attempts to automatially renew the certificate, the new certificate is created but not applied across the nodes in the respective machine config pool. This causes failure in multiple `oc` commands, including but not limited to `oc debug`, `oc logs`, `oc exec`, and `oc attach`. Pausing an MCP should be done with careful consideration about the `kube-apiserver-to-kubelet-signer` CA certificate expiration and for short periods of time only.  
+====
+
+To ensure that the cluster remains stable, creating a custom MCP for the control plane nodes (also known as the master nodes) is not supported. The Machine Config Operator (MCO) ignores any custom MCP created for the control plane nodes.
+
+[id="update-using-custom-machine-config-pools-about-mcp_{context}"]
+== About the update process and MCPs
+
+In {product-title} nodes are not considered individually. Nodes are grouped into machine config pools(MCP). There two MCPs in a default {product-title} cluster: one for the control plane nodes and one for the worker nodes. An {product-title} update affects all MCPs concurrently.
+
+During the update, the Machine COnfig Operator (MCO) drains and cordons all nodes within a MCP up to the specified `maxUnavailable` number of nodes (if specified). Draining and cordoning a node deschedules all pods on the node and marks the node as unschedulable. Once drained the Machine Config Daemon applies a new machine configuration during the update, which can include updating the operating system, which requires the host to reboot. 
+
+To prevent specific nodes from being updated, and thus, not drained, cordoned, and updated, you can create custom MCPs to ensure that the nodes are not updated. The MCO does not update any paused MCPs. You can create one or more custom MCPs, which can give more control over the sequence in which nodes are updated. After the update of the first MCP, the application compatibility can be verfied and then the whole fleet can be updated gradually to the new version.
+
+In case of a failure, such as your applications not working on the updated nodes, you can cordon and drain the nodes in the pool, which moves the application pods to other nodes to help maintain the quality-of-service for the applications.
+
```

Comment:
I like that you mentioned `you can cordon and drain the nodes in the pool, which moves the application pods to other nodes to help maintain the quality-of-service for the applications.` here. But we still need the customer to think about the extra capacity and plan accordingly. Because that will help them to decide the pool size. Else if they realize that application is not working after updating 50% of the nodes that would be very bad for the application. The rest 50% most likely wont be able to give same level of QOS. 

Here is the text from my PR. 
```
. Create MCPs out of the worker pool. The number of nodes in each MCP depends on a few factors like maintaince window duration for each MCP and the amount of reserve capacity (extra worker nodes) present.
+
[NOTE]
====
In case of a failure i.e. if the MCP with the new version does not work as expected with the applications, the nodes in the pool can be cordoned and drained. So that the application will not run on these nodes and the extra capacity will help to maintain the quality of service of the applications.
```

Reply:
@LalatenduMohanty What do you mean by extra capacity or reserve capacity?

Reply:
@mburke5678 for example if the cluster consists of 100 nodes and there are 2 MCP. If the customer realizes after updating one MCP that the application is not working on the updated MCP then the application has only 50 nodes to run its pods. The question  is this enough for the application to maintain the quality of service. Most likely not. The customer most likely has to add more new worker nodes to reach the quality of service. Thats the reserve capacity.